src/transform/src/canonicalization/flat_map_elimination.rs

```diff
@@ -7,16 +7,19 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
-//! Turns `FlatMap` into `Map` if only one row is produced by flatmap.
-//!
+//! For a `FlatMap` whose args are all constants, turns it into `Map` if only 1 row is produced by
+//! the table function, or turns it into an empty constant if 0 rows are produced by the table
```

Comment:
"empty constant" probably should be "empty collection"?

Reply:
Changed to "empty constant collection"